operator/pkg/manifest/shared.go

```diff
@@ -523,20 +529,24 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 		p = strings.TrimPrefix(p, "spec.")
 		inc, _, err := tpath.GetPathContext(iop, util.PathFromString("spec."+p), true)
 		if err != nil {
-			return "", err
+			return "", "", err
 		}
 		// input value type is always string, transform it to correct type before setting.
 		if err := tpath.WritePathContext(inc, util.ParseValue(v), false); err != nil {
-			return "", err
+			return "", "", err
 		}
 	}
 
 	out, err := yaml.Marshal(iop)
 	if err != nil {
-		return "", err
+		return "", "", err
+	}
+	manifests := strings.Split(iopYAML, "---")
```

Comment:
we are looking for `\n---` .  a `---` in the middle of the yaml file is still fine.